File: wellness/wellness.Model/Report/Report.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace wellness.Model.Report
{
    public class Report
    {
        public int Id { get; set; }
        public DateTime DateTo { get; set; }
        public DateTime DateFrom { get; set; }

        public float EarnedMoney { get; set; }
        public DateTime Timestamp { get; set; }

        public int TotalUsers { get; set; }
        public string MemberShipTypeName { get; set; } = null!;
        public int MemberShipTypeId { get; set; }
    }
}

File: wellness/UI/desktop/lib/models/report.dart
import 'package:json_annotation/json_annotation.dart';

part 'report.g.dart';

@JsonSerializable()
class Report {
  final int id;
  final DateTime dateTo;
  final DateTime dateFrom;
  final double earnedMoney;
  final String? memberShipTypeName;
  final int memberShipTypeId;
  final int? totalUsers;

  factory Report.fromJson(Map<String, dynamic> json) => _$ReportFromJson(json);

  Report(this.id, this.dateTo, this.dateFrom, this.earnedMoney,
      this.memberShipTypeName, this.memberShipTypeId, this.totalUsers);

  Map<String, dynamic> toJson() => _$ReportToJson(this);
}
